master/pkg/model/defaults.go

```diff
@@ -29,6 +29,13 @@ const (
 	defaultGPUImage = "determinedai/environments:cuda-10.2-pytorch-1.7-tf-1.15-gpu-1c4ea10"
 )
 
+const (
+	// DefaultCheckpointStorageType is SharedFS.
+	DefaultCheckpointStorageType = "shared_fs"
+	// DefaultSharedFSHostPath is the default path on hosts for SharedFS storage mounts.
+	DefaultSharedFSHostPath = "/tmp"
```

Comment:
I think adding this default will mean users can now not supply `checkpoint_storage` anywhere and have a valid exp conf, and this is new afaik, it always had to be set somewhere previously. Now that there is a valid default i'd make sure the defaulting logic with the master config's checkpoint config and the template checkpoint configs doesn't get busted. I'm not sure right off what would happen.

Reply:
If the flag is going to apply to the master config, and det deploy just needs to edit the master config defaults, do we need this code messing with the exp conf at all?

Reply:
1. I assume previously it'd come from the pre-packaged master.yaml, which explicitly specifies /tmp. I suppose the change is that if you pass a master.yaml which doesn't have checkpoint_storage, we'd now store data in /tmp instead of exiting. That probably won't break existing users.
2. There's a unit test `TestUnmarshalMasterConfigurationViaViper`, which effectively checks that the config values are the same as in `internal.DefaultConfig`. All of this helps to make that true.

Reply:
Yeah, but this change updates the default experiment config (`model.DefaultExperimentConfig`), not master config (`internal.DefaultConfig`).

Reply:
master's checkpoint storage config comes from experiment config: https://github.com/ioga/determined/blob/master/master/internal/config.go#L30
do you think default checkpoint storage config should be separate and different for master and experiment configs?

Reply:
yeah i think that makes sense. I think experiments should have no default but master config should have a default (or have no default and have det-deploy supply the default for local cases).

Reply:
went down the latter path of not changing the default, please see the updated code.

It doesn't use `registerString` machinery and doesn't expose the same config as a flag, because `pflags` methods require you to provide the default, so using `BindPFlag` (e.g. in `registerString`) ends up setting the default. And it can't be empty string either, because the parser throws up because it's not a valid config.

Reply:
sweet, thanks for updating.